src/main/java/com/fzi/externalcontrol/impl/ExternalControlInstallationNodeContribution.java

```diff
@@ -169,10 +175,35 @@ public void onOk(String value) {
   }
 
   public String getUrScriptProgram() {
-	if (urScriptProgram == "") {
-	  RequestProgram sender = new RequestProgram(getHostIP(), getCustomPort());
-	  urScriptProgram = sender.sendCommand("request_program\n");
-	}
-    return urScriptProgram;
+    return urScriptControlLoop;
+  }
+
+  public void findHeaderAndControlLoop(String script) {
+    int it = 0;
+    Boolean controlLoopFound = false;
+    String[] splitData = script.split("\n");
+
+    while (it < splitData.length) {
+      // Find the first line of the control loop
+      if (splitData[it].startsWith(" socket_open(")) {
```

Comment:
This seems absolutely brittle. As you mentioned in the OP, it might be better to define some anchors.

Reply:
Once the script has been moved to the client library [issue](https://github.com/UniversalRobots/Universal_Robots_Client_Library/issues/31) I will define some anchors to search for.